### PR TITLE
Pcode parser rewrite

### PR DESCRIPTION
**Pull request by m-rtz**
Next step of pcode parser rewrite.

IR assigns supported added for unary operation, binary operations, casts and subpiece.

### CODE REVIEW

**Review by Enkelmann — CHANGES_REQUESTED** [review]
Only 2 minor comments, rest looks good.

File: src/cwe_checker_lib/src/ghidra_pcode/pcode_op_simple/mod.rs [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+                        address: address.to_string(),
+                    },
+                    term: load2,
+                });
+
+                // Change varnode to virtual register
+                self.input0.id = "$load_temp1".into();
+                self.input0.address_space = "unique".into();
+            }
+        }
```
[the remark this answers is not in the document]
Comment (m-rtz) [reply src/cwe_checker_lib/src/ghidra_pcode/pcode_op_simple/mod.rs]: Good point!

File: src/cwe_checker_lib/src/ghidra_pcode/pcode_operations.rs [hunk 2]
```diff
@@ -94,3 +94,56 @@ impl ExpressionType {
...
+        if matches!(self, JmpType::CALL) {
+            Jmp::CallOther {
+                description,
+                return_,
+            }
+        } else {
+            panic!("Not a call operation")
+        }
+    }
+}
```
Comment (Enkelmann) [inline src/cwe_checker_lib/src/ghidra_pcode/pcode_operations.rs:149]: Is this a typo that you test for `JmpType::CALL` instead of `CALLOTHER`?

Comment (m-rtz) [reply src/cwe_checker_lib/src/ghidra_pcode/pcode_operations.rs:149]: Good catch!

**Pull request merged**